api/src/main/java/org/openmrs/api/impl/UserServiceImpl.java

```diff
@@ -760,5 +762,16 @@ public User setUserActivationKey(User user) {
 		return user;
 	}
 
+	/**
+	 * @see org.openmrs.api.UserService#changeUserPasswordUsingActivationKey(String, String);
+	 */
+	@Override
+	public void changeUserPasswordUsingActivationKey(String activationKey, String newPassword) throws APIException {
+		User user = getUserByActivationKey(activationKey);
+		if (user == null) {
+			throw new APIException("activation.key.not.correct", (Object[]) null);
```

Comment:
Why are you passing a null value in as an argument? You might as well pass in no argument

Reply:
This is a comparison not a parameter passing